### DISCUSSION TITLE
Table extraction returning None for certain column in only some rows

### DISCUSSION THREAD

**Discussion by MatinQurban** [jsvine/pdfplumber#1175] [Ask for help with specific PDFs]
Hello, I am extracting data from a pdf I am not sure why it returns None 

<img width="474" alt="image" src="https://github.com/user-attachments/assets/a4bb689d-e9bd-42a6-925a-bae954afc895">
Here is the table.debug

Here is an example of the data I get:
['Estimate Totals', 'Discount $', 'Markup $', None, 'Rate $', 'Total Hours', 'Total $']
['Parts', '', '66.00', None, '', '', '2,490.48']
['Labor, Body', '', '', None, '70.00', '61.2', '4,284.00']
['Labor, Refinish', '', '', None, '70.00', '21.6', '1,512.00']
['Labor, Mechanical', '', '', None, '165.00', '3.1', '511.50']
['Material, Paint', '', '', None, '', '', '1,036.80']
['Miscellaneous', '', '', None, '', '', '171.80']
['Subtotal', '', '', None, '', '', '10,006.58']
['Sales Tax', '', '', None, '', '', '281.18']
['Grand Total', '', '', None, '', '', '10,287.76']
['Deductible', '', '', None, '', '', '(1,000.00)']
['Net Total', '', '', None, '', '', '9,287.76']

I tried it with another pdf of the same format and got this:
<img width="482" alt="image" src="https://github.com/user-attachments/assets/f6587d3b-c277-45ae-9594-2312e6f71e12">

Data:
['Estimate Totals', 'Discount $', 'Markup $', 'Rate $', 'Total Hours', 'Total $']
['Parts', '', '121.25', '', '', '1,428.50']
['Labor, Body', '', '', '73.00', '15.9', '1,160.70']
['Labor, Refinish', '', '', '73.00', '10.2', '744.60']
['Labor, Mechanical', '', '', '198.99', '1.0', '198.99']
['Feather Prime and Block', '', '', '73.00', '1.2', '87.60']
['Material, Paint', '', '', '', '', '510.00']
['E.P.C.', '', '', '', '', '5.00']

Found a post that talks about using extract_text and making sure it works with pdfminer.six but since I'm using tables I thought it might be a different issue. Any ideas?

**Comment by jsvine** [comment]
Hi @MatinQurban, and thanks for your interest in `pdfplumber`. The library's table-extraction output represents tables as lists (rows) of lists (cell values). The representation is straightforward for tables that have consistent rows and columns. But some tables have more complex structures, such as nesting or grouped headers. In those cases, some rows will not "intersect" with all columns, and instead us `None` to represent the non-values at those non-intersections. In your first screenshot, you can see that there's a sub-table being recognized (toward the bottom) within the main table, creating such a situation.

If your PDFs format such tables in a consistent manner, you might have some luck first using `page.crop(...)` to focus just one sub-table at a time.
